[jit][edge] Use dynamic type instead of union types for schema parsers.

zhxchen17 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #70509
* #70233
* #70212
* #70205

TypeFactory will construct DynamicType when building on Edge platforms. We use this facility to make FunctionSchema return DynamicType all the time for OptionalType. We don't explicitly use DynamicTypeFactory everywhere because that requires too many changes and will split the entire aten codebase.

Differential Revision: [D33306737](https://our.internmc.facebook.com/intern/diff/D33306737/)

zhxchen17 committed:
[jit][edge] Use dynamic type instead of union types for schema parsers.

TypeFactory will construct DynamicType when building on Edge platforms. We use this facility to make FunctionSchema return DynamicType all the time for OptionalType. We don't explicitly use DynamicTypeFactory everywhere because that requires too many changes and will split the entire aten codebase.

Differential Revision: [D33306737](https://our.internmc.facebook.com/intern/diff/D33306737/)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Use dynamic type instead of union types for schema parsers."


TypeFactory will construct DynamicType when building on Edge platforms. We use this facility to make FunctionSchema return DynamicType all the time for OptionalType. We don't explicitly use DynamicTypeFactory everywhere because that requires too many changes and will split the entire aten codebase.

Differential Revision: [D33306737](https://our.internmc.facebook.com/intern/diff/D33306737/)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Use dynamic type instead of union types for schema parsers."


TypeFactory will construct DynamicType when building on Edge platforms. We use this facility to make FunctionSchema return DynamicType all the time for OptionalType. We don't explicitly use DynamicTypeFactory everywhere because that requires too many changes and will split the entire aten codebase.

Differential Revision: [D33306737](https://our.internmc.facebook.com/intern/diff/D33306737/)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Use dynamic type instead of union types for schema parsers."


TypeFactory will construct DynamicType when building on Edge platforms. We use this facility to make FunctionSchema return DynamicType all the time for OptionalType. We don't explicitly use DynamicTypeFactory everywhere because that requires too many changes and will split the entire aten codebase.

Differential Revision: [D33306737](https://our.internmc.facebook.com/intern/diff/D33306737/)

[ghstack-poisoned]

iseeyuan reviewed (approved):
LGTM. Could you add the build size result in the comment of this PR? Thanks!

zhxchen17 committed:
Update on "[jit][edge] Use dynamic type instead of union types for schema parsers."


TypeFactory will construct DynamicType when building on Edge platforms. We use this facility to make FunctionSchema return DynamicType all the time for OptionalType. We don't explicitly use DynamicTypeFactory everywhere because that requires too many changes and will split the entire aten codebase.

Differential Revision: [D33306737](https://our.internmc.facebook.com/intern/diff/D33306737/)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Use dynamic type instead of union types for schema parsers."


TypeFactory will construct DynamicType when building on Edge platforms. We use this facility to make FunctionSchema return DynamicType all the time for OptionalType. We don't explicitly use DynamicTypeFactory everywhere because that requires too many changes and will split the entire aten codebase.

Differential Revision: [D33306737](https://our.internmc.facebook.com/intern/diff/D33306737/)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Use dynamic type instead of union types for schema parsers."


TypeFactory will construct DynamicType when building on Edge platforms. We use this facility to make FunctionSchema return DynamicType all the time for OptionalType. We don't explicitly use DynamicTypeFactory everywhere because that requires too many changes and will split the entire aten codebase.

Differential Revision: [D33306737](https://our.internmc.facebook.com/intern/diff/D33306737/)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Use dynamic type instead of union types for schema parsers."


TypeFactory will construct DynamicType when building on Edge platforms. We use this facility to make FunctionSchema return DynamicType all the time for OptionalType. We don't explicitly use DynamicTypeFactory everywhere because that requires too many changes and will split the entire aten codebase.

Differential Revision: [D33306737](https://our.internmc.facebook.com/intern/diff/D33306737/)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Use dynamic type instead of union types for schema parsers."


TypeFactory will construct DynamicType when building on Edge platforms. We use this facility to make FunctionSchema return DynamicType all the time for OptionalType. We don't explicitly use DynamicTypeFactory everywhere because that requires too many changes and will split the entire aten codebase.

Differential Revision: [D33306737](https://our.internmc.facebook.com/intern/diff/D33306737/)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Use dynamic type instead of union types for schema parsers."


TypeFactory will construct DynamicType when building on Edge platforms. We use this facility to make FunctionSchema return DynamicType all the time for OptionalType. We don't explicitly use DynamicTypeFactory everywhere because that requires too many changes and will split the entire aten codebase.

Differential Revision: [D33306737](https://our.internmc.facebook.com/intern/diff/D33306737/)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Use dynamic type instead of union types for schema parsers."


TypeFactory will construct DynamicType when building on Edge platforms. We use this facility to make FunctionSchema return DynamicType all the time for OptionalType. We don't explicitly use DynamicTypeFactory everywhere because that requires too many changes and will split the entire aten codebase.

Differential Revision: [D33306737](https://our.internmc.facebook.com/intern/diff/D33306737/)

[ghstack-poisoned]
